staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/azure.go

```diff
@@ -180,6 +180,7 @@ type azureToken struct {
 
 type tokenSource interface {
 	Token() (*azureToken, error)
+	refreshToken(*azureToken) (*azureToken, error)
```

Comment:
I think it is better to be a public function. also can you rename it to `Refresh(*azureToken) (*azureToken, error)`